packages/compat/src/compat-app.ts

```diff
@@ -146,6 +147,16 @@ export default class CompatApp {
     }
   }
 
+  @Memoize()
+  private get contentFor(): ContentForConfig {
+    const configPath = join('environments', `${this.legacyEmberAppInstance.env}.json`);
```

Comment:
It feels like the test failures are related to this line. The code comes from `htmlTree` getter, but `testIndexTree` getter has:
```
  configPath: join('environments', `test.json`)
```
might be related.